File: typescript/src/services/groups-api.ts
/** Generated API client. */
import type { ApiResponse } from '../types.js';
import type { DocmostHttpClient } from '../http-client.js';
import type { AddGroupUserDto, CreateGroupDto, GroupIdDto, PaginationOptions, RemoveGroupUserDto, UpdateGroupDto } from '../models/index.js';


export class GroupsApi {
  constructor(private readonly http: DocmostHttpClient) {}

  async getWorkspaceGroups(request: PaginationOptions): Promise<ApiResponse<unknown>> {
    return this.http.post(`groups`, request);
  }

  async getGroupInfo(request: GroupIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`groups/info`, request);
  }

  async createGroup(request: CreateGroupDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`groups/create`, request);
  }

  async updateGroup(request: UpdateGroupDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`groups/update`, request);
  }

  async getGroupMembers(): Promise<ApiResponse<unknown>> {
    return this.http.post(`groups/members`, undefined);
  }

  async addGroupMembers(request: AddGroupUserDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`groups/members/add`, request);
  }

  async removeGroupMember(request: RemoveGroupUserDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`groups/members/remove`, request);
  }

  async deleteGroup(request: GroupIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`groups/delete`, request);
  }

}

File: c#/src/Docmost.Sdk/Services/GroupsApi.cs
using Docmost.Sdk.Http;
using Docmost.Sdk.Models;
using System.Net.Http.Headers;
using System.Net.Http.Json;
using System.Text.Json;

namespace Docmost.Sdk.Services;

public sealed class GroupsApi(DocmostHttpClient http)
{
    private readonly DocmostHttpClient _http = http;

    public async Task<ApiResponse<JsonElement?>> GetWorkspaceGroupsAsync(PaginationOptions request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("groups", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetGroupInfoAsync(GroupIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("groups/info", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> CreateGroupAsync(CreateGroupDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("groups/create", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> UpdateGroupAsync(UpdateGroupDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("groups/update", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetGroupMembersAsync(CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("groups/members", null, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> AddGroupMembersAsync(AddGroupUserDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("groups/members/add", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> RemoveGroupMemberAsync(RemoveGroupUserDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("groups/members/remove", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> DeleteGroupAsync(GroupIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("groups/delete", request, cancellationToken).ConfigureAwait(false);
    }

}
